src/routing.test.js

```diff
@@ -0,0 +1,59 @@
+import { parsePath } from './routing'
+import { APP_MODE_START, APP_MODE_ORG } from './symbols'
+
+const ADDRESS = '0xc41e4c10b37d3397a99d4a90e7d85508a69a5c4c'
+
+function locator(path, data) {
+  return {
+    path,
+    pathname: path,
+    search: '',
+    ...data,
+  }
+}
+
+describe('parsePath()', () => {
+  test('it does something', () => {
+    let path
+
+    path = '/'
+    expect(parsePath(path)).toEqual(
+      locator(path, {
+        mode: APP_MODE_START,
+      })
+    )
+
+    path = `/p/${ADDRESS}/test`
+    expect(parsePath(path)).toEqual(
+      locator(path, {
+        dao: 'p.aragonid.eth',
+        instanceId: ADDRESS,
+        instancePath: '/test',
+        mode: APP_MODE_ORG,
+        params: null,
+      })
+    )
+
+    path = `/p/${ADDRESS}`
+    expect(parsePath(path)).toEqual(
+      locator(path, {
+        dao: 'p.aragonid.eth',
+        instanceId: ADDRESS,
+        instancePath: '/',
+        mode: APP_MODE_ORG,
+        params: null,
+      })
+    )
+
+    path = `/p/${ADDRESS}///`
```

Comment:
Is this a valid path?

Reply:
I don’t think anyone will use a slash followed by another one, but I don’t see a reason to forbid it. Do you think it should be the case?